### PR TITLE
Bug 1215482 - Apply the new bookmarks design to TV Browser.

### PR DESCRIPTION
**Pull request by weilonge**

### CODE REVIEW

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=74f1628a06d9dc2dbf39b61e1ee33ca08cb002c2

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=92e34f113aa98f4fae43b22ae16433b2fc7d1a60

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=00953d77f1add554e9dd2c360978b1dba1970f87

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=4e493563492aecbe6b56d0debabd65ef39de710a

File: build/config/common-settings.json [hunk 1]
```diff
@@ -235,7 +235,7 @@
    "support.callsupport1.href": "",
    "support.callsupport2.title": "",
    "support.callsupport2.href": "",
-   "sync.collections.history.enabled": true,
+   "sync.collections.history.enabled": false,
```
Comment (michielbdejong) [inline build/config/common-settings.json]: Should not be part of this patch, probably?

Comment (weilonge) [inline build/config/common-settings.json]: We will do that in build/config/tv/settings.json .

File: tv_apps/browser/js/datasync/sync_bookmarks.js [hunk 2]
```diff
@@ -11,23 +11,38 @@
...
-      var data;
-      if (task.data) {
-        data = task.data.fxsyncPayload;
-        data.timestamp = task.data.last_modified;
-      }
-      switch(task.operation) {
-      case 'update':
-        SyncBrowserDB.updateBookmark(data, () => {
+      if (data.deleted) {
+        console.log(data); // XXX
```
Comment (michielbdejong) [inline tv_apps/browser/js/datasync/sync_bookmarks.js]: Please remove console.log statements.

File: tv_apps/smart-system/js/scratchpad.js [hunk 3]
```diff
@@ -0,0 +1,638 @@
+'use strict';
```
Comment (michielbdejong) [inline tv_apps/smart-system/js/scratchpad.js]: What does this file do? Is it meant to go into master, given its name?

File: tv_apps/browser/js/datasync/sync_bookmarks.js [hunk 4]
```diff
@@ -11,23 +11,38 @@
...
+      }
+      switch(task.operation) {
+      case 'update':
+      case 'add':
+        //Promise.all(data.map(updateWrapper)).then(resolve);
+        data.reduce((reduced, current) => {
+          return reduced.then(() => {
+            return updateWrapper(current);
+          });
+        }, Promise.resolve()).then(resolve);
```
Comment (michielbdejong) [inline tv_apps/browser/js/datasync/sync_bookmarks.js]: Nicer code pattern would be `Promise.all(data.map(fxSyncRecord => updateWrapper(fxSyncRecord))`

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=0c3987bef69d4150e7844c38b2d509f2426033d4

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=1bdf197aea11533b15233640a131822c549960f9

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=623c20dbff06ec203ce8829970a6a520706863bc
